File: DiagonalDifference.cs
class Result
{

    /*
     * Complete the 'diagonalDifference' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts 2D_INTEGER_ARRAY arr as parameter.
     */

    public static int diagonalDifference(List<List<int>> arr)
    {
        var left = 0;
        var right = 0;
        
        for (var i = 0; i < arr[0].Count(); i++)
        {
            left += arr[i][i];
            right += arr[i][arr[0].Count() - 1 - i];
        }
        
        return Math.Abs(left - right);
    }

}

File: HackerRank/1 WeeK Preparation Kit/DiagonalDifference.py
# Given a square matrix, calculate the absolute difference between the sums of its diagonals.
# For example, the square matrix  is shown below:

# 1 2 3
# 4 5 6
# 9 8 9  

# reuslt is 2

def diagonalDifference(arr):
    left = 0
    right = 0

    for i in range(len(arr[0])):
        left = left + arr[i][i]
        right = right + arr[i][len(arr[0]) - i - 1]

    return abs(left - right)
